firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Browser/SearchEngines/SearchEngineSelectionMiddlewareTests.swift

```diff
@@ -8,41 +8,61 @@ import XCTest
 @testable import Client
 
 final class SearchEngineSelectionMiddlewareTests: XCTestCase {
+    var mockStore: MockStoreForMiddleware<AppState>!
     var mockProfile: MockProfile!
-    var mockSearchEngines: [OpenSearchEngine]!
+    var mockSearchEnginesManager: SearchEnginesManager!
+    let mockSearchEngines: [OpenSearchEngine] = [
+        OpenSearchEngineTests.generateOpenSearchEngine(type: .wikipedia, withImage: UIImage()),
+        OpenSearchEngineTests.generateOpenSearchEngine(type: .youtube, withImage: UIImage()),
+    ]
+    var mockSearchEngineModels: [SearchEngineModel] {
+        return mockSearchEngines.map({ $0.generateModel() })
+    }
 
     override func setUp() {
         super.setUp()
         DependencyHelperMock().bootstrapDependencies()
         mockProfile = MockProfile()
-        mockSearchEngines = [
-            OpenSearchEngineTests.generateOpenSearchEngine(type: .wikipedia, withImage: UIImage()),
-            OpenSearchEngineTests.generateOpenSearchEngine(type: .youtube, withImage: UIImage()),
-        ]
+        mockSearchEnginesManager = SearchEnginesManager(prefs: mockProfile.prefs, files: mockProfile.files)
+        mockSearchEnginesManager.orderedEngines = mockSearchEngines
+
+        // We must reset the global mock store prior to each test
+        mockStore = MockStoreForMiddleware(state: AppState())
+        store = mockStore
     }
 
     override func tearDown() {
         DependencyHelperMock().reset()
         super.tearDown()
     }
 
-    func testDismissMenuAction() throws {
-        let mockSearchEnginesManager = SearchEnginesManager(prefs: mockProfile.prefs, files: mockProfile.files)
-        mockSearchEnginesManager.orderedEngines = mockSearchEngines
-
+    func testViewDidLoad_dispatchesDidLoadSearchEngines() throws {
         let subject = createSubject(mockSearchEnginesManager: mockSearchEnginesManager)
         let action = getAction(for: .viewDidLoad)
 
-        let testStore = Store(
-            state: AppState(),
-            reducer: AppState.reducer,
-            middlewares: [subject.searchEngineSelectionProvider]
-        )
+        subject.searchEngineSelectionProvider(AppState(), action)
+
+        guard let actionCalled = mockStore.dispatchCalled.withActions.first as? SearchEngineSelectionAction,
+              case SearchEngineSelectionActionType.didLoadSearchEngines = actionCalled.actionType else {
+            XCTFail("Unexpected action type dispatched")
+            return
+        }
```

Comment:
hmm 🤔, wondering if we can simplify this some how so we don't have to do this check here for each action. Can think this through later in our meeting today

Reply:
yes, will think about this 🤔